src/python/pants/backend/python/packaging/pyoxidizer/BUILD

```diff
@@ -2,3 +2,25 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 python_sources()
+
+python_tests(
+    name="config_test",
+    sources=["config_test.py"],
+)
+
+python_tests(
+    name="rules_integration_test",
+    sources=["rules_integration_test.py"],
+    timeout=240,
+    interpreter_constraints=[">=3.9"]
+)
+
+python_tests(
+    name="subsystem_integration_test",
+    sources=["subsystem_integration_test.py"],
+)
+
+python_tests(
+    name="target_types_integration_test",
+    sources=["target_types_integration_test.py"],
+)
```

Comment:
Fly by comment, a more idiomatic pattern is to use a single `python_tests` target with default `sources`, along with `overrides` for `rules_integration_test`. It results in less boilerplate.

Tip that `update-build-files` is nice to handle formatting of the BUILD file via Black, `overrides` is annoying to manually format

Reply:
Thanks @Eric-Arellano! I was actually copying from another plugin - in my own projects, I only ever have a single `python_test` per repo, but I assumed there was some reason for this layout regarding the pants test infra.

Reply:
> but I assumed there was some reason for this layout regarding the pants test infra.

Only that we haven't finished updating all our BUILD files to use the best idiom. "Generated targets" have been a thing since Pants 2.0, but they were in the shadows -- you could only explicitly create a `python_test` target + use `overrides` starting in Pants 2.8. Before then, the BUILD file you have here is roughly the best we could get.

I have a WIP PR to update them that I sometimes work on when I need a trivial thing to do as a break from other things 😀